ui/app/AppLayouts/Chat/stores/UsersStore.qml

```diff
@@ -3,9 +3,21 @@ import QtQuick 2.15
 QtObject {
     id: root
 
+    property var chatCommunitySectionModule
+    property var chatDetails
     property var usersModule
 
-    readonly property var usersModel: usersModule ? usersModule.model : null
+    readonly property var usersModel: {
+        if (!chatDetails && !chatCommunitySectionModule) {
+            return null
+        }
+        let isFullCommunityList = !chatDetails.requiresPermissions
+        if (chatDetails.belongsToCommunity && isFullCommunityList && !!chatCommunitySectionModule) {
+            // Community channel with no permisisons. We can use the section's membersModel
+            return chatCommunitySectionModule.membersModel
+        }
+        return usersModule ? usersModule.model : null
+    }
```

Comment:
I moved the logic from ChatView to this store and reused the store whenever we need to get the right member model